src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.IncrementalAnalyzerProcessor.cs

```diff
@@ -182,12 +190,9 @@ private void ResetLogAggregator()
                     Func<IIncrementalAnalyzer, T, CancellationToken, Task> runnerAsync,
                     CancellationToken cancellationToken)
                 {
-                    // this is a best effort progress report. since we don't clear up
-                    // reported progress when work is done or cancelled. it is possible
-                    // that last reported work is left in report even if it is already processed.
-                    // but when everything is finished, report should get cleared
-                    _ = _registration.ProgressReporter.Update(GetFilePath(value));
+                    using var evaluating = _registration.ProgressReporter.GetEvaluatingScope();
 
+                    ReportPendingWorkItemCount();
```

Comment:
💡 If I'm not mistaken, `GetEvaluatingScope` will always trigger an event which in turn resets the delay. This pending work item count will therefore _never_ immediately update the UI. You could instead reverse the order of these operations, with the pending work item count update first (where it's more likely to trigger an update) followed by the evaluating scope change (which will force the update but only when needed).

Reply:
resetting delay is for the last event. not for limiting interval.

wrote explanation here.
https://github.com/dotnet/roslyn/pull/37943/files#diff-fecb712d7896a08890d74844af27cb4cR32

Reply:
This comment is unrelated to the interval limit, and describes a specific bug in the current implementation.